packages/promises/_test.pony

```diff
@@ -145,3 +146,22 @@ class iso _TestPromisesJoinThenReject is UnitTest
     a("a")
     b("b")
     c.reject()
+
+class iso _TestNextUnwrap is UnitTest
+  fun name(): String => "promises/Promise.next_unwrap"
+
+  fun apply(h: TestHelper) =>
+    let start = Promise[String]
+    let inter = start.next_unwrap[String](_NextUnwrapHelper~promise_string())
+    inter.next[None](_NextUnwrapHelper~complete(h))
+
+    start("start")
+    inter("end")
+
+primitive _NextUnwrapHelper
+  fun promise_string(s: String): Promise[String] =>
+    // fake do something with string, return promise of another
+    Promise[String]
+
+  fun complete(h: TestHelper, s: String) =>
+    h.assert_eq[String]("end", s)
```

Comment:
I see an assertion for "end" here, but not one for "start" -- should a "start" assertion be added?

Reply:
Start exists entirely so we can have something to call `next_unwrap` on. Think testing its value in `promise_string` makes the case that we are testing less clear.

Reply:
Fair enough. My rationale was from a perspective of "tests as documentation" to show where in the process "start" is accessible. For the same reason of "tests as documentation" I agree that as it is makes it clear where "end" is accessible -- which is the additional functionality here.